packages/ui-extensions/src/surfaces/customer-account/targets.ts

```diff
@@ -230,3 +230,37 @@ export interface FullPageApi {
 export interface ActionExtensionApi {
   close(): void;
 }
+
+/**
+ * For a given extension target, returns the type of the API that the
+ * extension will receive at runtime.
+ */
+export type ApiForExtension<Target extends keyof ExtensionTargets> =
+  ExtractedApiFromExtension<ExtensionTargets[Target]>;
+
+type ExtractedApiFromRunExtension<T> = T extends RunExtension<
+  infer Api,
+  unknown
+>
+  ? Api
+  : never;
+
+type ExtractedApiFromExtension<T> = T extends RenderExtension<any, any>
```

Comment:
I tried this out in hub. Also, there will [be another pr](https://github.com/Shopify/customer-account-web/pull/2908) on hub to make sure hub and and 1p extensions work fine together.

Since this pr will not impact anything in prod, thus it is safe. @robin-drexler